contracts/Manager.sol

```diff
@@ -722,97 +429,179 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
+     * @param beneficiary The address that should end up receiving the withdrawn
+     * CELO.
+     **/
+    function distributeWithdrawals(uint256 stCeloAmount, address beneficiary) private {
+        uint256 withdrawal = toCelo(stCeloAmount);
+        if (withdrawal == 0) {
+            revert ZeroWithdrawal();
+        }
+
+        address strategy = _checkAndUpdateStrategy(beneficiary, strategies[beneficiary]);
+
+        address[] memory groupsWithdrawn;
+        uint256[] memory withdrawalsPerGroup;
+
+        if (strategy != address(0)) {
+            (groupsWithdrawn, withdrawalsPerGroup) = specificGroupStrategy
+                .calculateAndUpdateForWithdrawal(strategy, withdrawal, stCeloAmount);
         } else {
-            emit GroupRemoved(group);
+            (groupsWithdrawn, withdrawalsPerGroup) = defaultStrategy
+                .calculateAndUpdateForWithdrawal(withdrawal);
         }
+
+        account.scheduleWithdrawals(beneficiary, groupsWithdrawn, withdrawalsPerGroup);
     }
 
     /**
-     * @notice Checks if a group member is elected.
-     * @param groupMember The member of the group to check election status for.
-     * @return Whether or not the group member is elected.
+     * @notice Checks if strategy was deprecated. Deprecated strategy is reverted to default.
+     * Updates the strategies.
+     * @param accountAddress The account.
+     * @param strategy The strategy.
+     * @return Up to date strategy
      */
-    function isGroupMemberElected(address groupMember) private view returns (bool) {
-        IElection election = getElection();
-
-        address[] memory electedValidatorSigners = election.electValidatorSigners();
+    function _checkAndUpdateStrategy(address accountAddress, address strategy)
+        private
+        returns (address)
+    {
+        address checkedStrategy = _checkStrategy(strategy);
+        if (checkedStrategy != strategy) {
+            strategies[accountAddress] = checkedStrategy;
+        }
+        return checkedStrategy;
+    }
 
-        for (uint256 i = 0; i < electedValidatorSigners.length; i++) {
-            if (electedValidatorSigners[i] == groupMember) {
-                return true;
-            }
+    /**
+     * @notice Schedules transfer of Celo between strategies.
+     * @param fromStrategy The from validator group.
+     * @param toStrategy The to validator group.
+     * @param stCeloAmount The stCelo amount.
+     * @param from The from address.
+     * @param to The to address.
+     */
+    function _transfer(
+        address fromStrategy,
+        address toStrategy,
+        uint256 stCeloAmount,
+        address from,
+        address to
+    ) private {
+        fromStrategy = _checkAndUpdateStrategy(from, fromStrategy);
+        toStrategy = _checkAndUpdateStrategy(to, toStrategy);
+
+        if (fromStrategy == toStrategy) {
+            // either both addresses use default strategy
+            // or both addresses use same allowed strategy
```

Comment:
Should references to "allowed strategy" be changed to "specific strategy" throughout?